usagov/usagov-2021
# USAGOV-1198: add menu entries for directory exports

## Jira Task

https://cm-jira.usa.gov/browse/USAGOV-1198

## Description

Adds some menu entries to the admin area for content editors. This includes a route and a controller for the root of the new "USAGov tools" menu, and menu links for two recently created views. 

(Why the usa_admin_styles module? I wanted to put these in a place that wasn't specific to certain features -- like the directory records. Even though this module has "styles" in its name, it seems like a good fit: it's specific to the admin area and how we display things there.)  

## Type of Changes

  - Add screenshot showing what it should look like
- [x] Other - Routing, requires flushing the routing and links cache. 

## Testing Instructions



### Requires New Config

### Requires New Content

### Validation Steps
After you've installed this and flushed the routing and links cache (`bin/drush cr` will do that and more), you should see a new **USAGov tools** menu when you're logged in. A site admin will see this: 

![Screenshot 2024-02-14 at 6 23 17 PM](https://github.com/usagov/usagov-2021/assets/35591/5b582d9e-18a9-494f-88b2-5ddd336a6322)

You should also be able to see this menu when logged in as a user who has only a Content role. I recommend that you pick an appropriate user from the People tab, edit their profile, and add a password so you can log in as that user. It will look like this: 

![Screenshot 2024-02-14 at 6 28 13 PM](https://github.com/usagov/usagov-2021/assets/35591/a79f96f1-3f1e-49db-9044-01608b7434bc)

Selecting either report from the menu will download a CSV file. That's not actually part of this work; it was already in place, but there was no link to it. 

Finally, clicking on "USAGov tools" will bring up a page with the same links that are in the menu. I don't love it, but there has to be a page and that is it. 

## Security Review




## Reviewer Reminders
- Reviewed code changes
- Reviewed functionality
- Security review complete or not required

## Post PR Approval Instructions
Follow these steps as soon as you merge the new changes.

1. Go to the [USAGov Circle CI project](https://app.circleci.com/pipelines/github/usagov/usagov-2021).
2. Find the commit of this pull request.
3. Build and deploy the changes.
4. Update the Jira ticket by changing the ticket status to `Review in Test` and add a comment. State whether the change is already visible on [cms-dev.usa.gov](http://cms-dev.usa.gov/) and [beta-dev.usa.gov](http://beta-dev.usa.gov/), or if the deployment is still in process.
